### ISSUE TITLE
Score is linked to online "Guest" user when logging in after a play is started

### ISSUE THREAD

**Issue by Juvice9362** [ppy/osu#25039]
### Type

Game behaviour

### Bug description

I opened osu! lazer and spammed the space bar which brought me straight into a randomly selected map. While the map was on the loading screen, the client prompted me to log in, which I did, and I continued with the play. When I finished the map and went to check my local scores, I found a score under the username "Guest" with a profile attached to it. This is not an account that I have ever interacted with.
This is my first bug report please let me know if there is anything missing.

### Screenshots or videos

![osu_2023-10-07_01-26-35](https://github.com/ppy/osu/assets/147228651/3fd5b6fe-a649-4a8c-947c-1798c40a5315)
![osu_2023-10-07_01-22-33](https://github.com/ppy/osu/assets/147228651/e46dda29-12bb-424a-83ed-e7928cf3c173)
![osu_2023-10-07_01-22-35](https://github.com/ppy/osu/assets/147228651/e4590d63-24d5-4f59-82a0-d6d57fe5ed54)
![osu_2023-10-07_01-22-45](https://github.com/ppy/osu/assets/147228651/d824b17a-2391-4aaf-b5d6-049f3bdaabf1)
![osu_2023-10-07_01-23-07](https://github.com/ppy/osu/assets/147228651/d37ba3b4-a1fb-452c-82e1-8ebe9aa18d1a)

### Version

2023.924.1-lazer

### Logs

[performance.log](https://github.com/ppy/osu/files/12836867/performance.log)
[runtime.log](https://github.com/ppy/osu/files/12836868/runtime.log)
[updater.log](https://github.com/ppy/osu/files/12836869/updater.log)
[database.log](https://github.com/ppy/osu/files/12836870/database.log)
[input.log](https://github.com/ppy/osu/files/12836871/input.log)
[legacy-ipc.log](https://github.com/ppy/osu/files/12836872/legacy-ipc.log)
[network.log](https://github.com/ppy/osu/files/12836873/network.log)

**Comment by bdach** [comment]
> While the map was on the loading screen, the client prompted me to log in

Can you elaborate on this? I'm not aware of any mechanism that would do this.

**Comment by Juvice9362** [comment]
I opened up the game and I was spamming the space bar. This is equivalent to pressing "select" very quickly so the game selected "play", "solo", then opened up the randomly selected beat map chosen when loading into the menu for the first time in a session. It did this process in a very short amount of time. I do not know if this is relevant but it is all that happened before I was brought to the screen in the screenshot where I was prompted to log in because I had been logged out in my last session. 

Usually, the game brings me to the login prompt screen in the top right when I open up the game after being logged out. It does this in the Menu screen when booting up the game. The page looked like the last screenshot right before I began to play.

My user name read "Guest" at the top as it usually does when I am logged out in lazer. After logging in, I set a play and checked my locals and the score appears as if it was set by the user with the username "Guest", and account that has never been on my computer or I have ever interacted with. 
![osu_2023-10-07_02-52-17](https://github.com/ppy/osu/assets/147228651/bad3d2de-7498-42ae-aba8-61abbaf0c891)
![osu_2023-10-07_02-52-47](https://github.com/ppy/osu/assets/147228651/01a1e9c3-5ab0-4f42-8e46-812cf935bc88)
![osu_2023-10-07_02-57-59](https://github.com/ppy/osu/assets/147228651/dc4af97f-6e4c-429e-bf45-124d91f9a46c)
